src/ln/channelmanager.rs

```diff
@@ -3441,6 +3441,82 @@ mod tests {
+
+			nodes[0].chain_monitor.block_connected_with_filtering(&Block { header, txdata: vec![revoked_local_txn[0].clone()] }, 1);
+			test_txn_broadcast(&nodes[0], &chan_1, Some(revoked_local_txn[0].clone()), HTLCType::TIMEOUT);
+
+			nodes[1].chain_monitor.block_connected_with_filtering(&Block { header, txdata: vec![revoked_local_txn[0].clone()] }, 1);
+			let node_txn = nodes[1].tx_broadcaster.txn_broadcasted.lock().unwrap();
+			assert_eq!(node_txn.len(), 4); // ChannelManager is broadcasting 2 last_local_commitment_txn..and revocation tx is broacasted twice as block has been re-scanned.
+			assert_eq!(node_txn[0].input.len(), 2);
+
+			let mut funding_tx_map = HashMap::new();
+			funding_tx_map.insert(revoked_local_txn[0].txid(), revoked_local_txn[0].clone());
+			node_txn[0].verify(&funding_tx_map).unwrap();
+		}
+		get_announce_close_broadcast_events(&nodes, 0, 1);
+		assert_eq!(nodes[0].node.list_channels().len(), 0);
+		assert_eq!(nodes[1].node.list_channels().len(), 0);
+
+		let chan_2 = create_announced_chan_between_nodes(&nodes, 2, 3);
+
+		// Rebalance the network to generate htlc in the two directions
+		send_payment(&nodes[2], &vec!(&nodes[3])[..], 8000000);
+		// node[2] is gonna to revoke an old state thus node[3] should be able to claim both offered/received HTLC outputs on top of commitment tx, but this
+		// time as two different claim transactions as we're gonna to timeout htlc with given a high current heigh
+		let payment_preimage_3 = route_payment(&nodes[2], &vec!(&nodes[3])[..], 3000000).0;
+		let _payment_preimage_4 = route_payment(&nodes[3], &vec!(&nodes[2])[..], 3000000).0;
+
+		// Get the will-be-revoked local txn from node[2]
+		let revoked_local_txn = nodes[2].node.channel_state.lock().unwrap().by_id.iter().next().unwrap().1.last_local_commitment_txn.clone();
+
+		//Revoke the old state
+		claim_payment(&nodes[2], &vec!(&nodes[3])[..], payment_preimage_3);
+
+		{
+			let header = BlockHeader { version: 0x20000000, prev_blockhash: Default::default(), merkle_root: Default::default(), time: 42, bits: 42, nonce: 42 };
+
+			nodes[2].chain_monitor.block_connected_with_filtering(&Block { header, txdata: vec![revoked_local_txn[0].clone()] }, 200);
+			test_txn_broadcast(&nodes[2], &chan_2, Some(revoked_local_txn[0].clone()), HTLCType::TIMEOUT);
+
+			nodes[3].chain_monitor.block_connected_with_filtering(&Block { header, txdata: vec![revoked_local_txn[0].clone()] }, 200);
+			let node_txn = nodes[3].tx_broadcaster.txn_broadcasted.lock().unwrap();
+			assert_eq!(node_txn.len(), 10); // ChannelManager : 2, ChannelMontitor: 8 (2 revocation htlc tx, 1 local commitment tx + 1 htlc timeout tx) * 2 (block-rescan)
+			assert_eq!(node_txn[0].input.len(), 1);
+			assert_eq!(node_txn[1].input.len(), 1);
+
+			let mut funding_tx_map = HashMap::new();
+			funding_tx_map.insert(revoked_local_txn[0].txid(), revoked_local_txn[0].clone());
```

Comment:
Hmm, this test looks like it would pass whether the tx here were claiming the revoked normal output or the revoked HTLC output. We should test that later txn claim a different output in the revoked_local_txn[0], at least, if not be more aggressive with what all of the txn here look like.

Reply:
Yes add assert based on witness_script. Note: we can't assume which one is the htlc offered/received first as of using HashMap internally.